libflux/go/libflux/buffer.go

```diff
@@ -0,0 +1,45 @@
+package libflux
+
+// #cgo CFLAGS: -I.
+// #cgo LDFLAGS: -L. -lflux
+// #include "flux.h"
+// #include <stdlib.h>
+import "C"
+
+import (
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	flatbuffers "github.com/google/flatbuffers/go"
+)
+
+type ManagedBuffer struct {
+	Buffer []byte
+	Offset flatbuffers.UOffsetT
+	freeFn func()
+}
+
+func (b *ManagedBuffer) Free() {
+	if b.freeFn != nil {
+		b.freeFn()
+		b.freeFn = nil
+	}
+}
+
+func unwrapBuffer(buf C.struct_flux_buffer_t) *ManagedBuffer {
```

Comment:
`newManagedBuffer` would be a nice name for this that would be a little more Go-idiomatic.